packages/openapi-v3/src/__tests__/unit/decorators/request-body/spike.request-body.decorator.unit.ts

```diff
@@ -0,0 +1,161 @@
+import { getControllerSpec, post, put } from '../../../..';
+import { TS_TYPE_KEY } from '../../../../controller-spec';
+import { newRequestBody1 } from '../../../../decorators/request-body.option1.decorator';
+
+describe.only('spike - requestBody decorator', () => {
+  context('proposal 1', () => {
+    @model()
+    class Product extends Entity {
+      @property({
+        type: 'string',
+      })
+      name: string;
+      @belongsTo(() => Category)
+      categoryId: number;
+
+      constructor(data?: Partial<Product>) {
+        super(data);
+      }
+    }
+
+    /**
+    * Navigation properties of the Product model.
+    */
+    interface ProductRelations {
+      category?: CategoryWithRelations;
+    }
+    /**
+    * Product's own properties and navigation properties.
+    */
+    type ProductWithRelations = Product & ProductRelations;
+
+    @model()
+    class Category extends Entity {
+      @hasMany(() => Product)
+      products?: Product[];
+    }
+    /**
+    * Navigation properties of the Category model.
+    */
+    interface CategoryRelations {
+      products?: ProductWithRelations[];
+    }
+    /**
+    * Category's own properties and navigation properties.
+    */
+    type CategoryWithRelations = Category & CategoryRelations;
+
+    it('create - generates schema with excluded properties', () => {
+      const requestBodySpec = {
+        description: 'Create a product',
+        required: true,
+      };
+
+      const excludeOptions = {
+        [TS_TYPE_KEY]: Product,
+        schemaName: 'ProductWithoutID',
+        exclude: ['id']
+      }
+
+      class MyController1 {
+        @post('/Product')
+        create(@newRequestBody1(
+          requestBodySpec,
+          excludeOptions
+        ) product: Exclude<Product, ['id']>) { }
+      }
+
+      const spec1 = getControllerSpec(MyController1)
+
+      const requestBodySpecForCreate = spec1.paths[
+        '/Product'
+      ]['post'].requestBody;
+
+      const referenceSchema = spec1.components!.schemas!.ProductWithoutID;
+
+      expect(requestBodySpecForCreate).to.have.properties({
+        description: 'Create a product',
+        required: true,
+        content: {
+          'application/json': {
+            schema: {
+              $ref: '#/components/schemas/ProductWithoutID'
+            }
+          }
+        }
+      });
+
+      // The feature that generates schemas according to
+      // different options is TO BE DONE and out of the
+      // scope of this spike, so that the schema `PartialProduct`
+      // here is still the same as `Product`
+      expect(referenceSchema).to.have.properties({
+        title: 'ProductWithoutID',
+        properties: {
+          categoryId: { type: 'number' },
+          name: { type: 'string' }
+        }
+      });
+    })
+
+    it('update - generates schema with partial properties', () => {
+      const requestSpecForUpdate = {
+        description: 'Update a product',
+        required: true,
+      };
```

Comment:
I find this approach, where we need one new variable (`requestSpecForUpdate`, `requestSpecForCreate`)  for each endpoint accepting a request body, as not very ergonomic.

Please update [examples/todo/src/controllers/todo.controller.ts](https://github.com/strongloop/loopback-next/blob/extract/relation-helpers/examples/todo/src/controllers/todo.controller.ts) to show your proposal in practice.